spec/syntax.md

```diff
@@ -669,6 +651,45 @@ reserved-char = %x00-08        ; omit HTAB and LF
               / %xE000-10FFFF
 ```
 
+## Markup
+
+**_<dfn>Markup</dfn>_** _placeholders_ are _pattern_ parts
+that can be used to represent non-language parts of a _message_,
+such as inline elements or styling that should apply to a span of parts.
+Markup comes in three forms:
+
+**_<dfn>Markup-open</dfn>_** starts with `#` and
+is a _pattern_ part starting a span.
+It MAY include _options_, but these are not required.
+
+If a _markup-open_ includes a `/` immediately before its closing `}`,
+it is a self-closing **_<dfn>markup-standalone</dfn>_** part
+representing a standalone element within a message,
+such as an inline image or any other opaque content.
+
+**_<dfn>Markup-close</dfn>_** starts with `/` and
+is a _pattern_ part ending a span.
+Unlike the other forms, it does not include _options_.
+
+```abnf
+markup       = "{" [s] markup-open [s] ["/"] "}"
+             / "{" [s] markup-close [s] "}"
+markup-open  = "#" identifier *(s option)
+markup-close = "/" identifier
+```
+
+> A _message_ with one `button` markup span and a standalone `img` markup element.
+>
+> ```
+> {#button}Submit{/button} or {#img alt=|Cancel| /}.}
+> ```
+
+_Markup-open_ and _markup-close_ parts SHOULD be paired,
```

Comment:
Isn't that more of a `MAY`? There exist completely valid systems where markup is stack-based (e.g. TTY) and where I could imagine only `markup-open` being used with success.

Reply:
```suggestion
_Markup-open_ and _markup-close_ parts MAY be paired,
```